lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -1960,9 +1981,8 @@ static void insertSRSOp(Operation *Op, VectState *state) {
   if (Op->use_empty() || Op->getNumResults() == 0)
     return;
 
-  // The operation must be in AIE dialect, and its result should be accumulator
-  // type.
-  assert(isAIEOp(Op) && llvm::any_of(Op->getResultTypes(), isAccType));
+  // The operation must write to an accumulator
+  assert(writesToAccumulator(Op));
 
   // Check if any user of this operation is a non-AIE op. If any user of this
   // operation is non-AIE op, then we need to generate SRS op to move value
```

Comment:
This is where it's determined that SRS is inserted only when we go from accumulator to non-AIEVec op